invisible_cities/detsim/light_tables_c.pyx

```diff
@@ -0,0 +1,171 @@
+        self.max_psf     = max(lt_df.index.values)
+        self.max_psf2    = self.max_psf**2
+        self.nsensors    = len(self.sensor_ids_)
+
+
+    @cython.boundscheck(False)
+    @cython.wraparound(False)
+    @cython.cdivision(True)
+    @cython.initializedcheck(False)
+    cdef double* get_values_(self, const double x, const double y, const int sns_id):
+        cdef:
+            double dist
+            double aux
+            unsigned int psf_bin_id
+            double xsipm
+            double ysipm
+            double tmp_x
+            double tmp_y
+            double*  values
+        if sns_id >= self.nsensors:
+            return NULL
+        xsipm = self.snsx[sns_id]
+        ysipm = self.snsy[sns_id]
+        tmp_x = x-xsipm; tmp_y = y-ysipm
+        dist = tmp_x*tmp_x + tmp_y*tmp_y
+        if dist>self.max_psf2:
+            return NULL
+        if x*x+y*y>=self.active_r2:
+            return NULL
+        aux = sqrt(dist)*self.inv_bin
+        bin_id = <int> floor(aux)
+        values = &self.values[bin_id, 0]
+        return values
+
+
+cdef class LT_PMT(LT):
+    cdef:
+        double [:, :, :, ::1] values
+        double max_zel
+        double max_psf
+        double max_psf2
+        double inv_binx
+        double inv_biny
+        double xmin
+        double ymin
+        double active_r2
+
+    def __init__(self, *, fname, el_gap=None, active_r=None):
+        from scipy.interpolate import griddata
+        lt_df, config_df, el_gap, active_r = read_lt(fname, 'LT', el_gap, active_r)
+        self.el_gap   = el_gap
+        self.active_r = active_r
+        self.active_r2 = active_r**2
+
+        sensor = config_df.loc["sensor"].value
+        columns = [col for col in lt_df.columns if ((sensor in col) and ("total" not in col))]
+        el_pitch    = el_gap #this is hardcoded for this specific table, should this be in config?
+        self.zbins_ = get_el_bins(el_pitch, el_gap)
+
+        self.sensor_ids_ = np.arange(len(columns)).astype(np.intc)
+        xtable   = lt_df.x.values
+        ytable   = lt_df.y.values
+        xmin_, xmax_ = xtable.min(), xtable.max()
+        ymin_, ymax_ = ytable.min(), ytable.max()
+        bin_x = float(config_df.loc["pitch_x"].value) * units.mm
+        bin_y = float(config_df.loc["pitch_y"].value) * units.mm
+        # extend min, max to go over the active volume
+        xmin, xmax = xmin_-np.ceil((self.active_r-np.abs(xmin_))/bin_x)*bin_x, xmax_+np.ceil((self.active_r-np.abs(xmax_))/bin_x)*bin_x
+        ymin, ymax = ymin_-np.ceil((self.active_r-np.abs(ymin_))/bin_y)*bin_y, ymax_+np.ceil((self.active_r-np.abs(ymax_))/bin_y)*bin_y
+        #create new centers
+        x          = np.arange(xmin, xmax+bin_x/2., bin_x).astype(np.double)
+        y          = np.arange(ymin, ymax+bin_y/2., bin_y).astype(np.double)
+        xx, yy     = np.meshgrid(x, y)
+        values_aux = (np.concatenate([griddata((xtable, ytable), lt_df[column], (yy, xx), method='nearest')[..., None]
```

Comment:
This bit could be moved to a separate member function, adding a couple of lines documenting what you're doing (something like "Extend light table to fully cover the chosen radius extrapolating from the existing data.").